core/consensus/pow/primitives/src/lib.rs

```diff
@@ -20,17 +20,46 @@
 
 use rstd::vec::Vec;
 use sr_primitives::ConsensusEngineId;
+use codec::Decode;
+use substrate_client::decl_runtime_apis;
 
 /// The `ConsensusEngineId` of PoW.
 pub const POW_ENGINE_ID: ConsensusEngineId = [b'p', b'o', b'w', b'_'];
 
-/// Type of difficulty.
-///
-/// For runtime designed for Substrate, it's always possible to fit its total
-/// difficulty range under `u128::max_value()` because it can be freely scaled
-/// up or scaled down. Very few PoW chains use difficulty values
-/// larger than `u128::max_value()`.
-pub type Difficulty = u128;
-
 /// Type of seal.
 pub type Seal = Vec<u8>;
+
+/// Define methods that total difficulty should implement.
+pub trait TotalDifficulty {
+	fn add(&mut self, other: Self);
+}
+
+impl TotalDifficulty for primitives::U256 {
+	fn add(&mut self, other: Self) {
+		let ret = self.saturating_add(other);
+		*self = ret;
+	}
+}
+
+impl TotalDifficulty for u128 {
+	fn add(&mut self, other: Self) {
```

Comment:
`fn increment`?

Reply:
Hehe, see [this](https://github.com/paritytech/substrate/pull/3617#discussion_r330491441). My suggestion was `increase` but I think `increment` should be also good